reco_utils/dataset/movielens.py

```diff
@@ -128,50 +135,50 @@ def item_has_header(self):
 ERROR_MOVIE_LENS_SIZE = "Invalid data size. Should be one of {100k, 1m, 10m, or 20m}"
 ERROR_LOCAL_CACHE_PATH = """Local cache path only accepts a zip file path:
     use/something/like_this.zip"""
+ERROR_NO_HEADER = "No header (schema) information"
 
 
 def load_pandas_df(
     size="100k",
-    header=(
-        DEFAULT_USER_COL,
-        DEFAULT_ITEM_COL,
-        DEFAULT_RATING_COL,
-        DEFAULT_TIMESTAMP_COL
-    ),
-    local_cache_path="ml.zip",
+    header=DEFAULT_HEADER,
+    local_cache_path=None,
     title_col=None,
     genres_col=None,
+    year_col=None,
 ):
     """Loads the MovieLens dataset as pd.DataFrame.
 
     Download the dataset from http://files.grouplens.org/datasets/movielens, unzip, and load
 
     Args:
-        size (str): Size of the data to load. One of ("100k", "1m", "10m", "20m")
-        header (list or tuple or None): Rating dataset header. If None, ratings are not loaded.
-        local_cache_path (str): Path where to cache the zip file locally
-        title_col (str): Movie title column name. If None, the title column is not loaded.
+        size (str): Size of the data to load. One of ("100k", "1m", "10m", "20m").
+        header (list or tuple or None): Rating dataset header.
+        local_cache_path (str): Path where to cache the zip file locally.
+            If None, use default, which is "data/ml-{size}/ml.zip"
+        title_col (str): Movie title column name. If None, the column will not be loaded.
         genres_col (str): Genres column name. Genres are '|' separated string.
-            If None, the genres column is not loaded.
+            If None, the column will not be loaded.
+        year_col (str): Movie release year column name. If None, the column will not be loaded.
 
     Returns:
         pd.DataFrame: Movie rating dataset.
-            If header is None but either title_col or genres_col is not None,
-            returns movie titles and/or genres.
     """
+    movie_col = DEFAULT_ITEM_COL if header is None or len(header) < 2 else header[1]
 
     # fix capitalization
     size = size.lower()
 
-    datapath, item_datapath = _load_datafile(size, local_cache_path)
+    datapath, _ = _maybe_download_and_extract(size, local_cache_path)
 
-    # Load title and genres
-    movie_col = DEFAULT_ITEM_COL if header is None or len(header) < 2 else header[1]
-    item_df = _load_item_df(size, movie_col, title_col, genres_col, item_datapath)
+    # Load movie features such as title, genres, and release year
+    if title_col is None and genres_col is None and year_col is None:
+        item_df = None
+    else:
+        item_df = load_item_df(size, local_cache_path, movie_col, title_col, genres_col, year_col)
 
     # Load rating data
     if header is None or len(header) == 0:
```

Comment:
can we move this check to the beginning, no point in downloading anything if the header is invalid

Reply:
Good catch. Should've moved up after the refactor.